### PR TITLE
Changed to pointer method receiver

### PR DESCRIPTION
**Pull request by italolelis**
## What does this PR do?

 As @endeveit mentioned before it's better using pointer method receivers than literals, so this aims to fix that.

### CODE REVIEW

File: middleware_oauth.go [hunk 1]
```diff
@@ -3,17 +3,18 @@ package main
 import (
 	"encoding/json"
 
+	"errors"
+	"net/http"
```
Comment (endeveit) [inline middleware_oauth.go:7]: Good that you grouped your imports in blocks like `stdlib`, `external dependencies`, `project packages` separated with empty line 👍 
But it looks like you forgot to remove empty line on line 5

File: middleware_oauth2_key_exists.go [hunk 2]
```diff
@@ -4,17 +4,18 @@ import (
 	"errors"
 	"strings"
 
+	"net/http"
```
Comment (endeveit) [inline middleware_oauth2_key_exists.go:7]: And also here

File: middleware_oauth.go (hunk 1 above)
Comment (italolelis) [reply middleware_oauth.go:7]: Actually the empty line is from golint, it puts there automatically

**Comment by wellfreire** [comment]
+1

[Approved with PullApprove](https://pullapprove.com/hellofresh/api-gateway/pull-request/9/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=hellofresh/api-gateway)

File: middleware_oauth.go (hunk 1 above)
Comment (endeveit) [reply middleware_oauth.go:7]: hm, ok then, but its looks strange for me 🤔

Comment (endeveit) [reply middleware_oauth.go:7]: Yeah! That is exactly what I'm talking about.

> Imports are organized in groups, with blank lines between them. The standard library packages are in the first group.

And you have an empty line between "encoding/json" and "errors". They are both from standard library

**Pull request merged**